compiler/rustc_hir_typeck/src/method/suggest.rs

```diff
@@ -1597,7 +1597,11 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             .filter(|item| matches!(item.kind, ty::AssocKind::Fn) && !item.fn_has_self_parameter)
             .filter_map(|item| {
                 // Only assoc fns that return `Self`, `Option<Self>` or `Result<Self, _>`.
-                let ret_ty = self.tcx.fn_sig(item.def_id).skip_binder().output();
+                let ret_ty = self
+                    .tcx
+                    .fn_sig(item.def_id)
+                    .instantiate(self.tcx, self.fresh_args_for_item(DUMMY_SP, item.def_id))
+                    .output();
```

Comment:
```suggestion
                    .output()
                    .instantiate(self.tcx, self.fresh_args_for_item(DUMMY_SP, item.def_id));
```

Reply:
EarlyBinder<FnSig>::output() doesnt exist

Reply:
Oh heh. I was looking at polyfnsig

Reply:
yeah I think youre thinking of the late binder, not early binder haha